node/connection.js

```diff
@@ -213,6 +223,26 @@ TChannelConnection.prototype.onCallResponse = function onCallResponse(res) {
     req.responseEvent.emit(req, res);
 };
 
+TChannelConnection.prototype.onPingRequest = function onPingRequest(req) {
+    var self = this;
+    req.remoteAddr = self.remoteName;
```

Comment:
this can just be `handler.sendPingRequest()`, which has access to `self.hostPort`, which is the same as `self.remoteName` on conn.